src/sinks/splunk_hec/logs/config.rs

```diff
@@ -53,12 +52,12 @@ pub struct HecLogsSinkConfig {
 
     /// Overrides the name of the log field used to retrieve the hostname to send to Splunk HEC.
     ///
-    /// By default, the [global `log_schema.host_key` option][global_host_key] is used.
+    /// By default, the [global `log_schema.host_key` option][global_host_key] is used if log
+    /// events are Legacy namespaced, or the semantic meaning of "host" is used, if defined.
     ///
     /// [global_host_key]: https://vector.dev/docs/reference/configuration/global-options/#log_schema.host_key
     #[configurable(metadata(docs::advanced))]
-    #[serde(default = "config_host_key_target_path")]
-    pub host_key: OptionalTargetPath,
+    pub host_key: Option<OptionalTargetPath>,
```

Comment:
😵

Reply:
This is better than `Option<String>` because we can catch deserialization errors of the path creation at config initialization time and not have to do the parsing of the path multiple times.

Reply:
See the k8s logs source and the pulsar sink for other examples.

Reply:
🤔 This seems to indicate there is a modeling, or at least a naming, issue. It's probably outside of the scope of this PR, but I think it isn't clear when to use `Option<OptionalTargetPath>` and when to use `OptionalTargetPath`.

Reply:
:nods: yep, completely agree

Reply:
I'll open a task to track the modelling/naming issue

Reply:
Bruce also noticed this. I added some code comments in the interim.